homeassistant/components/nice_go/config_flow.py

```diff
@@ -29,6 +31,7 @@ class NiceGOConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Nice G.O."""
 
     VERSION = 1
+    reauth_entry: NiceGOConfigEntry | None = None
```

Comment:
```suggestion
    reauth_entry: NiceGOConfigEntry
```
Should only type it as otherwise you initialize it as a class variable that can be shared between reauth instances if one has more than one config entry.

We can even remove `None` as then we can skip the type check

Reply:
This prevents me from committing.

```
homeassistant/components/nice_go/config_flow.py:77: error: Incompatible types in assignment (expression has type "ConfigEntry[Any] | None", variable has type "ConfigEntry[NiceGOUpdateCoordinator]")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

Reply:
Right. Might need to type with `None` then and keep the assert

Reply:
This still needs to be addressed

Reply:
Sorry, I don't understand what needs to be addressed

Reply:
Remove ` = None` at the end to not initialize the variable.